### ISSUE TITLE
icon page includes (Default) in alt text for ALL icons

### ISSUE THREAD

**Issue by kareila** [dreamwidth/dw-free#1703]
Reported here: http://www.dreamwidth.org/support/see_request?id=32773

Verified on the reporting user's icons page which uses journal style, and on my own account's icons page which uses site style.

**Comment by pinterface** [comment]
Might as well claim this, since I'm staring at the code.

It seems to be an issue of LJ::Userpic->alttext always including "(Default)" (not translated, naturally) if a keyword string is not passed in, and LJ::S2::IconsPage not passing in a keyword string.  However, I'm not really clear on what the correct behavior /is/.

Near as I can puzzle out, LJ::Userpic->alttext was written with posts and comments in mind--that is, there's only one applicable userpic keyword, or there is none (fallback to default).

So what happens when that interacts with something like the /icons page is...well, things go weird.  "(Default)" in alt for every icon is only one example.  Another example would be the title attribute, which lists one and only one of the keywords which applies to that userpic.  If you're viewing by "Keyword Order", the keyword listed in title may not even be the same keyword the userpic is being displayed for (it'll be a different keyword also applied to that image).

There's also the autogenerated keywords (pic#$id) to be considered (filter out?  include?).  And possibly whether the information needs to be in the alt attribute at all: it's already next to the image on the page.

In other words, the desired behavior should probably be specced out.

**Comment by rahaeli** [comment]
Hmmm. My gut instinct is to say that yeah, alt and title should be blank (not missing, but "alt="" title=""") for the /icons page, since yeah, the information that would go into the alt/title is right there next to the icon on the page. I am completely willing to be overridden there, though, because I am not a screenreader user. @deborahgu -- thoughts? (Alternately, I can ping one of my screenreader-using friends to get her opinion.)

**Comment by rahaeli** [comment]
or, you know, I could ask dw-accessibility.
